### PR TITLE
yaml-dsl-autogen

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by davsclaus** [comment]
Ah this is great. So you find the model classes with those JAXB annotations, and then filter on label to get the "eip"'s and then skip some with has custom implementations, and otherwise generate for those as they are more straightforward and can be generated.

File: tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java [hunk 1]
```diff
@@ -144,4 +154,93 @@ public final TypeSpec generateReifiers() {
...
+                            switch (i.name().toString()) {
+                                case "org.apache.camel.model.Resilience4jConfigurationDefinition":
+                                case "org.apache.camel.model.HystrixConfigurationDefinition":
+                                case "org.apache.camel.model.config.StreamResequencerConfig":
+                                case "org.apache.camel.model.config.BatchResequencerConfig":
+                                case "org.apache.camel.model.OnFallbackDefinition":
+                                case "org.apache.camel.model.InOnlyDefinition":
+                                case "org.apache.camel.model.InOutDefinition":
+                                case "org.apache.camel.model.OtherwiseDefinition":
+                                case "org.apache.camel.model.WhenDefinition":
```
Comment (valdar) [inline tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java:209]: I know that this list is unlikely to frequently change, but it is also true that hardcoded here in quite easily missed if another custom implementation needs to be skipped.

Would it be good to have this configurable at the mojo level? Or documented somewhere?

File: tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java [hunk 2]
```diff
@@ -144,4 +154,93 @@ public final TypeSpec generateReifiers() {
...
+                                    return;
+                                default:
+                                    break;
+                            }
+                            switch (i.name().prefix().toString()) {
+                                case "org.apache.camel.model.loadbalancer":
```
[the remark this answers is not in the document]
Comment (lburgazzoli) [reply tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java]: done with https://github.com/apache/camel-k-runtime/pull/322/commits/e6c9522ac299cbc5f5ddd7f9bfd848d6e95343b9

File: tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java (hunk 1 above)
Comment (lburgazzoli) [reply tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java]: done with https://github.com/apache/camel-k-runtime/pull/322/commits/e6c9522ac299cbc5f5ddd7f9bfd848d6e95343b9

File: tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (valdar) [reply tooling/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateYamlLoaderSupportClasses.java]: LGTM

**Comment by lburgazzoli** [comment]
ok to test

**Pull request merged**